### ISSUE TITLE
Release dates/notes are inconsistent

### ISSUE THREAD

**Issue by cremor** [dotnet/core#9353]
### URL(s)

https://dotnet.microsoft.com/en-us/download/dotnet/8.0
https://github.com/dotnet/core/blob/main/release-notes/8.0/8.0.6/8.0.6.md
https://github.com/dotnet/core/blob/main/release-notes/8.0/8.0.6/8.0.302.md
https://github.com/dotnet/core/blob/main/release-notes/8.0/README.md

### Description

The [download page](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) currently shows:

> 8.0.6 Security patch
[Release notes](https://github.com/dotnet/core/blob/main/release-notes/8.0/8.0.6/8.0.6.md?WT.mc_id=dotnet-35129-website)
Latest release date    June 11, 2024

Problems I found:
- The linked release notes say ".NET 8.0.6 - May 28, 2024". This doesn't match the "June 11, 2024" from the downloads page.
- The version actually released on "June 11, 2024" was [.NET 8.0.302 -June 11, 2024](https://github.com/dotnet/core/blob/main/release-notes/8.0/8.0.6/8.0.302.md), but it looks like this is not linked from the download page (which makes sense, because it look like it doesn't actually contain any information on what changed in that version).
- Both release notes files say "The latest 8.0 release is always listed at [.NET 8.0 Releases](https://github.com/dotnet/core/blob/main/release-notes/8.0/README.md)." But the linked page doesn't contain version 8.0.6.
- The downloads page doesn't explain what the difference between SDK 8.0.3xx, SDK 8.0.2xx and SDK 8.0.1xx is. The release notes files do explain it (if you can spot the minor difference), but as noted above they are not linked on the downloads page.

**Comment by rbhanda** [comment]
https://github.com/dotnet/core/pull/9356 to fix the dates in readme files and adding cve informaiton

**Comment by reduckted** [comment]
There was also no "monthly update" post for June over in the [announcements](https://github.com/dotnet/announcements/issues?q=is%3Aissue+is%3Aopen+label%3AMonthly-Update) repository. Should there have been one?

**Comment by Falco20019** [comment]
Yep, neither here nor the blog post on https://devblogs.microsoft.com/dotnet/

**Comment by leecow** [comment]
@reduckted and @Falco20019 - There was no June monthly update release so nothing to report.

**Comment by reduckted** [comment]
> @reduckted and @Falco20019 - There was no June monthly update release so nothing to report.

😕 But 8.0.302 SDK was released in June.

**Comment by Falco20019** [comment]
Yes, 8.0.302 was the only June update but also https://github.com/dotnet/announcements/issues/306 is not linking the versions of https://github.com/dotnet/core/issues/9362 although mentioning that it should if there are more updates the month.

**Comment by cremor** [comment]
> The downloads page doesn't explain what the difference between SDK 8.0.3xx, SDK 8.0.2xx and SDK 8.0.1xx is. The release notes files do explain it (if you can spot the minor difference), but as noted above they are not linked on the downloads page.

I just found https://learn.microsoft.com/en-us/dotnet/core/versions/#versioning-details which explains this. That should be linked on the downloads page!

**Comment by mairaw** [comment]
@Falco20019 @cremor what would you like to see on the downloads page to help with that? A link to versioning details somewhere inside the table for each version?

**Comment by cremor** [comment]
I'd like to see a note that says something like the following:

"See the [versioning documentation](https://learn.microsoft.com/en-us/dotnet/core/versions/#versioning-details) to understand the different SDK versions. See [the release notes] for the feature differences."

"[the release notes]" should then link to a page that provides a nice overview of the feature changes between SDK 8.0.3xx, SDK 8.0.2xx and SDK 8.0.1xx.

**Comment by Falco20019** [comment]
I meant those mismatches on https://github.com/dotnet/announcements/issues/306:
![image](https://github.com/user-attachments/assets/f3db2b74-03f1-4910-9059-dc7b3634da1a)

https://github.com/dotnet/core/issues/9362
![image](https://github.com/user-attachments/assets/6bcf451a-dc92-48d3-aa57-460e29c1202a)

So the first is mentioning 8.0.4 and 8.0.5 while it should mention (and link) 8.0.6 instead.

**Comment by mairaw** [comment]
> I meant those mismatches on [dotnet/announcements#306](https://github.com/dotnet/announcements/issues/306):...

Tagging @rbhanda for the announcements descriptions that are incorrect.
